data_structures/linked_list/ascendingPriorityQueue.c

```diff
@@ -0,0 +1,271 @@
+/* Ascending priority queue using Linked List - Program to implement Ascending priority queue using Linked List */
+
+#include  <stdio.h>
+#include  <stdlib.h>  
+#define  NULL  0
+
+struct  node
+{
+  int  data ;
+  struct  node  *next ;
+} ;
+
+struct  node  *front , *rear ;
+
+/* This function initializes the queue to empty by making both front and rear as NULL */
+void createqueue()
+{
+  front=rear=NULL ;
+}
+
+int  empty()
+{
+  if(front==NULL)
+    return  1 ;
+  else
+    return  0 ;
+}
+
+void  insert(int  x)
+{
+  struct  node  *pnode ;
+
+  pnode=(struct node*)malloc(sizeof(struct node)) ;
+  if(pnode==NULL)
+  {
+    printf("Memory overflow. Unable to insert.\n") ;
+    getch();
+    exit(1) ;
+  }
+
+  pnode->data=x ;  
+  pnode->next=NULL;   /* New node is always last node */
+
+  if(empty())
+      front=rear=pnode ;
+  else
+  {
+      rear->next=pnode ;
+      rear=pnode ;
+  }
+}
+
+int  removes()
+{
+  int  min ;
+  struct  node  *follow , *follow1 , *p , *p1 ;
+
+  if(empty())
+  {
+    printf("\nQueue Underflow. Unable to remove.") ;
+    getch() ;
```

Comment:
`getch()` is a non-standard function and should be removed

Reply:
Done